specifyweb/frontend/js_src/lib/components/DataModel/resource.ts

```diff
@@ -225,12 +226,25 @@ export function resourceOn(
 export const parseJavaClassName = (className: string): string =>
   className.split('.').at(-1) ?? '';
 
+type InteractionTable = typeof interactionsWithPrepTables[number];
+const interactionTablesPrepsFieldName: RR<InteractionTable, string> = {
+  Disposal: 'disposalPreparations',
+  Loan: 'loanPreparations',
+  Gift: 'giftPreparations',
+  ExchangeOut: 'exchangeOutPreps',
+  ExchangeIn: 'exchangeInPreps',
+};
```

Comment:
could you please create this list dynamically?
like so:
```suggestion
const interactionTablesPrepsFieldName = f.store(()=>
  filterArray(
    Object.values(genericTables)
    .map((table=>{
      const relationshipToPreparation = table.relationships.find(({relatedTable})=>relatedTable === table.Preparation);
      return relationshipToPreparation ? [table.name, relationshipToPreparation.name] : undefined;
    })
  )
);
```